uniffi_bindgen/src/interface/mod.rs

```diff
@@ -1053,33 +1063,119 @@ impl APIConverter<Field> for weedle::dictionary::DictionaryMember<'_> {
         if let Type::Object(_) = type_ {
             bail!("Objects cannot currently appear in record fields");
         }
+        let default = match self.default {
+            None => None,
+            Some(v) => Some(v.value.convert(ci, &type_)?),
+        };
         Ok(Field {
             name: self.identifier.0.to_string(),
             type_,
             required: self.required.is_some(),
-            default: match self.default {
-                None => None,
-                Some(v) => Some(v.value.convert(ci)?),
-            },
+            default,
         })
     }
 }
 
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, Hash)]
+pub enum Radix {
+    Decimal = 10,
+    Octal = 8,
+    Hexadecimal = 16,
+}
+
 // Represents a literal value.
 // Used for e.g. default argument values.
 #[derive(Debug, Clone, Serialize, Deserialize, Hash)]
 pub enum Literal {
     Boolean(bool),
     String(String),
-    // TODO: more types of literal
+    UInt(u64, Radix, Type),
+    Int(i64, Radix, Type),
+    Float(String, Type),
```

Comment:
Naively, I was expecting to see separate `UInt8`, `UInt16` etc variants on this enum. I don't object to this more succinct representation if it makes the code cleaner, but it may be worth a short comment here to explain the choice since it stands out as being a bit different from other type-related enums.

Reply:
Added extra comments around these representations.

Reply:
Thanks for the extra comments! I also like your approach of using the widest representation possible, and truncating to the actual type in the binding backends, to simplify the code.